dask_sql/physical/rel/logical/sort.py

```diff
@@ -79,6 +73,12 @@ def _apply_sort(
             raise ValueError("Can not sort a column with NaNs")
 
         df = df.set_index(first_sort_column, drop=False).reset_index(drop=True)
+        if not first_sort_ascending:
+            # As set_index().reset_index() always sorts ascending, we need to reverse
+            # the order inside all partitions and the order of the partitions itself
+            df = dd.from_delayed(
+                reversed([self.reverse_partition(p) for p in df.to_delayed()])
+            )
```

Comment:
You may want to become familiar with the `df.partitions` iterable. 

```python
dd.concat(df.partitions[::-1])
```

Going between delayed and dataframes introduces some inconvenience in graph-handling.

Reply:
Uh nice! That looks awesome. Yes, this is definitely much better.

Reply:
At some point I think that we'll want to go through and remove all of the `to/from_delayed` calls.  This isn't critical, but it's nice to do from a performance standpoint (happy to go into this in more depth if you like).

Reply:
Thanks for your advice, I would actually be really interested in your thoughts about [this part](https://github.com/nils-braun/dask-sql/pull/75/files#diff-1fcfb9a4ff196f6a0e8fb3847cc3d2300b9b3a91af77f0555e6a61a179757802R153). 

Probably you see it right away, but what I would like to achieve is basically two things
1. call a function on every partition, while also having the partition index as an input argument
2. calculate `partition_borders`  only when executing the DAG - not already when creating it (this is what I have done before this PR, and it slows down the process).

So far, the `delayed` function was all I could come up with, but I definitely think there is a better way.

Reply:
I have created an issue to fix this, so we can go on with this PR.